akka-javasdk/src/main/scala/akka/javasdk/impl/agent/AgentImpl.scala

```diff
@@ -121,8 +121,9 @@ private[impl] final class AgentImpl[A <: Agent](
         commandEffect.primaryEffect match {
           case req: RequestModel =>
             val systemMessage = req.systemMessage match {
-              case ConstantSystemMessage(message)    => message
-              case TemplateSystemMessage(templateId) => promptTemplateClient.getPromptTemplate(templateId)
+              case ConstantSystemMessage(message) => message
+              case template: TemplateSystemMessage =>
+                promptTemplateClient.getPromptTemplate(template.templateId).format(template.args)
```

Comment:
use `.formatted` instead. Same thing, but unnecessary to go via Scala's implicit StringOps